components/text.js

```diff
@@ -178,6 +178,7 @@ export default memo(function Text ({ rel, imgproxyUrls, children, tab, itemId, o
             // If [text](url) was parsed as <a> and text is not empty and not a link itself,
             // we don't render it as an image since it was probably a conscious choice to include text.
             const text = children[0]
+            const internalURL = process.env.NODE_ENV === 'development' ? 'http://localhost:3000' : 'https://stacker.news'
```

Comment:
I would rather see a solution that uses `window.location.origin` but that will require use of `useEffect` due to SSR. Or a simple SSR check may be enough?

Reasons: not everyone runs their app on port 3000 or even on localhost and I would rather see less of these "prod or dev?"-checks than more. Related to this comment: https://github.com/stackernews/stacker.news/issues/1033#issuecomment-2041137657

Reply:
> Or maybe we should finally just export PUBLIC_URL to the client by prefixing it with NEXT_PUBLIC_

This could be helpful, but your brought up a good point and I think going with `window.location.origin` is probably best as I am frequently testing with ip on separate devices and whatnot

Reply:
@ekzyis do you think the coderabbit code here is sufficient to avoid needing a useEffect `const internalURL = typeof window === 'undefined' ? 'https://stacker.news' : window.location.origin` seems to pass the tests that get run on `/lib/url.js` just wanted to see what you think?

Reply:
Yes, looks good to me. Will throw hydration errors in local dev but I think it's better than `useEffect`.

_Btw, you can use the `SSR` constant from lib/constants.js, it does the same check._

Reply:
> Will throw hydration errors in local dev

Haven't been able to see where this might happen, or at least I can't produce this behavior